### PR TITLE
Fix symbolic shape function for `flatten` (silvasean's) 

### PR DESCRIPTION
**Pull request by makslevental**
in response to https://github.com/pytorch/pytorch/pull/71727

### CODE REVIEW

**Review by desertfire — APPROVED** [review]
Approve assuming CI failures will go away after #71727 is landed first.

**Comment by makslevental** [comment]
@desertfire 

> Approve assuming CI failures will go away after #71727 is landed first.

yea I wasn't expecting CI to fail. Will discuss with @eellison

**Pull request closed without merging**
